src/main/java/com/cburch/logisim/fpga/hdlgenerator/HDL.java

```diff
@@ -126,22 +128,20 @@ public static String xorOperator() {
   }
 
   public static String modOperator(String signalOne, String signalTwo, boolean signed) {
-    return (isVHDL()
-                ? "STD_LOGIC_VECTOR(" + (signed ? "SIGNED" : "UNSIGNED") + "(" + signalOne + ")"
-                : (signed ? "$signed(" + signalOne + ")" : signalOne))
+    return (isVHDL() ? "STD_LOGIC_VECTOR(" : "")
+            + typecast(signalOne, signed)
             + (isVHDL() ? " REM " : "%")
-            + (isVHDL()
-                ? (signed ? "SIGNED" : "UNSIGNED") + "(" + signalTwo + "))"
-                : (signed ? "$signed(" + signalTwo + ")" : signalTwo));
+            + typecast(signalTwo, signed)
+            + (isVHDL()? ")" : "");
   }
 
   public static String shiftlOperator(String signal, String nrOfBits, boolean arithmetic) {
-    return isVHDL() ? "STD_LOGIC_VECTOR(SHIFT_LEFT(" + (arithmetic ? "SIGNED(" : "UNSIGNED(") + signal + "), " + nrOfBits + "))"
+    return isVHDL() ? "STD_LOGIC_VECTOR(SHIFT_LEFT(" + typecast(signal, arithmetic) + ", " + nrOfBits + "))"
```

Comment:
Can we remove string concatenation in favor for `LineBuffer.format()` or even `String.format()` (multiple places)? I.e.

```java
public static String shiftlOperator(String signal, String nrOfBits, boolean arithmetic) {
    return isVHDL() 
        ? LineBuffer.format("STD_LOGIC_VECTOR( SHIFT_LEFT({{1}}, {{2}}) )", typecast(signal, arithmetic), nrOfBits)
        : LineBuffer.format("({{1}}{{2}}{{3}})", signal, (arithmetic ? " <<< " : " << "), nrOfBits);
}
```

In fact I should move this formatrer to separate class (or maybe to `S`?). But that's minor refactoring.